### PR TITLE
Extract model methods to services

### PR DESCRIPTION
**Pull request by jcoyne**
For publish, unpublish, revert, and purge.

This will allow us to move these services out of TuftsModels and into MIRA.

### CODE REVIEW

File: app/services/publish_service.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      obj.publishing = true
+      obj.save!
+      obj.publishing = false
+    end
+
+end
+
+class UnpublishableModelError < StandardError
+  def message
+    'Templates cannot be pushed to production'
```
Comment (tonyc) [inline app/services/publish_service.rb]: Change this to "Templates cannot be published" ?

File: app/models/concerns/publishable.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ module Publishable
 
   STATE_DELETED = 'D'
 
+  def publishable?
+    true
+  end
+
   def workflow_status
     raise "Production objects don't have a workflow" unless draft?
```
Comment (tonyc) [inline app/models/concerns/publishable.rb]: Change this to "Published objects don't have a workflow"

File: app/services/publish_service.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  def run
+    published_pid = PidUtils.to_published(object.pid)
+
+    destroy_published_version!
+    FedoraObjectCopyService.new(object.class, from: object.pid, to: published_pid).run
+
+    published = object.class.find(published_pid)
+    published!(published, user)
+    published!(object, user)
+    audit('Pushed to production')
```
Comment (tonyc) [inline app/services/publish_service.rb]: Change this to "Published" ?

File: app/services/publish_service.rb (hunk 1 above)
Comment (val99erie) [inline app/services/publish_service.rb]: This message should be changed.  Publishing no longer means 'push to production'.  How about something like this:  "#{object.class} cannot be published" ?

**Comment by tonyc** [comment]
:shipit:

**Pull request merged**